test/conftest.py

```diff
@@ -72,15 +64,19 @@ def _get_response(self, url: str, benchmark: Callable) -> Response:
         r = benchmark(lambda: urlopen(self.__base_url + url))
         return Response(status=r.status, headers=r.headers.items(), content=r.read())
 
-    def assert_404(self, url: str):
-        with pytest.raises(HTTPError) as e:
-            self._get_response(url, lambda f: f())
-            assert e.value.code == 404
-
-    def assert_403(self, url: str):
+    def assert_error_code(self, url: str, error_code: int):
         with pytest.raises(HTTPError) as e:
             self._get_response(url, lambda f: f())
-            assert e.value.code == 403
+        assert e is not None
+        error = e.value
+        assert_error_response(
+            Response(
+                status=error.code,
+                headers=error.headers.items(),
+                content=orjson_encode({"message": error.reason}),
```

Comment:
A bit artificial to reconstruct the content of `Response` here but as least, the error code is checked for.